src/System.Private.CoreLib/shared/Internal/Win32/RegistryKey.cs

```diff
@@ -440,7 +440,7 @@ internal void SetValue(string name, string value)
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
 
-            if (name != null && name.Length > MaxValueLength)
+            if (name.Length > MaxValueLength)
```

Comment:
This is internal and only being used here https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Environment.Win32.cs#L32

where we have Debug.Assert check to make sure name is not null

Reply:
While the check may not be necessary, I think there's benefit in keeping this code close to the public RegistryKey.SetValue version that's in corefx, which needs to keep the `name != null` check.  This isn't perf-critical, so I'd prefer to put back the `name != null &&` part.